File: IDE-Plugin-Sample/src/main/java/ui/ToolWindowFactory.java
package ui;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.ui.content.Content;
import com.intellij.ui.content.ContentFactory;
import org.jetbrains.annotations.NotNull;

public class ToolWindowFactory implements com.intellij.openapi.wm.ToolWindowFactory {

    @Override
    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        CustomToolWindow customToolWindow = new CustomToolWindow(toolWindow);
        ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
        Content content = contentFactory.createContent(customToolWindow.getContent(), "", false);
        toolWindow.getContentManager().addContent(content);
    }
}

File: IDE-Plugin-Sample/src/main/java/ui/tool_window/ToolWindowFactory.kt
package ui.tool_window

import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.ui.content.ContentFactory
import tooling.ContentRouterImpl

class ToolWindowFactory : com.intellij.openapi.wm.ToolWindowFactory {

    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        val router = ContentRouterImpl(toolWindow)

        val contentFactory = ContentFactory.SERVICE.getInstance()
        val configuringContent = contentFactory.createContent(ConfiguringContentView(project, router).contentPanel, "", false)
        val profilingContent = contentFactory.createContent(ProfilingContentView(router).contentPanel, "", false)

        router.setupContents(listOf(configuringContent, profilingContent))
        router.toNextContent()
    }
}